src/index.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable no-undef */
+const bookList = document.querySelector('#bookList');
+const dbref = firebase.database().ref();
+const booksref = dbref.child('books');
+
+function Book(id, author, title, pages, status) {
+  this.id = id;
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.readStatus = status;
+}
+
+function deleteBook({ target: { dataset: { id } } }) {
+  booksref.child(id).remove();
+  refresh();
+}
+
+function changeReadStatus(e) {
+  const value = e.target.dataset.value === 'read';
+  booksref.child(e.target.dataset.id).child('readStatus').set(value);
+  refresh();
+}
+
+function bookItem(book) {
+  const itemContainer = document.createElement('div');
+  itemContainer.setAttribute('class', 'col-lg-3 m-2');
+
+  const item = document.createElement('div');
+  item.classList.add('item');
+
+  const pages = document.createElement('span');
+  pages.setAttribute('class', 'pages-nbr px-3 pt-1 pb-2');
+  const pageText = document.createTextNode(`${book.pages} pages`);
+  pages.appendChild(pageText);
+
+  const icondiv = document.createElement('div');
+  icondiv.setAttribute(
+    'class',
+    'd-flex justify-content-center align-items-center my-5 book-icon',
+  );
+  const icon = document.createElement('i');
+  icon.setAttribute('class', 'ti-book');
+  icondiv.appendChild(icon);
+
+  const info = document.createElement('div');
+  info.setAttribute('class', 'px-3');
+
+  const title = document.createElement('h5');
+  title.setAttribute('class', 'title');
+  title.appendChild(document.createTextNode(`${book.title}`));
+
+  const author = document.createElement('span');
+  author.setAttribute('class', 'author');
+  author.appendChild(document.createTextNode(`» ${book.author}`));
+  info.appendChild(title);
+  info.appendChild(author);
+
+  const actions = document.createElement('div');
+  actions.setAttribute(
+    'class',
+    'actions d-flex mb-auto justify-content-between',
+  );
+
+  const readbtn = document.createElement('button');
+  readbtn.setAttribute('class', 'read');
+  readbtn.setAttribute('data-id', book.id);
+  const readbtnValue = book.readStatus === true ? 'not read' : 'read';
+  readbtn.setAttribute('data-value', readbtnValue);
+  readbtn.appendChild(document.createTextNode(readbtnValue));
+  readbtn.addEventListener('click', changeReadStatus);
+
+  const deletebtn = document.createElement('button');
+  deletebtn.setAttribute('class', 'delete');
+  deletebtn.setAttribute('data-id', book.id);
+  deletebtn.addEventListener('click', deleteBook);
+  deletebtn.appendChild(document.createTextNode('Delete'));
+
+  actions.appendChild(readbtn);
+  actions.appendChild(deletebtn);
+
+  item.appendChild(pages);
```

Comment:
- [ ] **[OPTIONAL]** We can increase the speed and performance of appending children to our nodes by avoiding reflows (drawing the page every time you append a child to a certain div) using [`createDocumentFragment`](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment) function. 

Think of it as a temporary element created in memory that you append children instead of the target element, and once you are done with appending any number of children, you append this fragment to the target without affecting the performance (one reflow vs multiple reflows).